modules/tensor_mechanics/include/materials/ADLAROMANCEStressUpdateBase.h

```diff
@@ -48,39 +50,35 @@ class ADLAROMANCEStressUpdateBase : public ADRadialReturnCreepStressUpdateBase
    */
   virtual Real computeTimeStepLimit() override;
 
+  /// Enum to error, warn, ignore, or extrapolate if input is outside of window of applicability
+  enum class WindowFailure
+  {
+    ERROR,
+    WARN,
+    IGNORE,
+    EXTRAPOLATE
+  };
+
   /**
    *Computes the ROM Strain rate
-   * @param dt Timestep size
-   * @param mobile_dislocations_old Mobile dislocation density from previous timestep
-   * @param immobile_dislocations_old Immobile dislocation density from previous timestep
-   * @param trial_stress Trial stress from radial return method
-   * @param effective_strain_old Effective strain from the previous timestep
-   * @param temperature Temperature variable value
-   * @param environmental Environmental variable value
    * @param mobile_dislocation_increment Mobile dislocation density incremental change
    * @param immobile_dislocation_increment Immobile dislocation density incremental change
    * @param rom_effective_strain ROM calculated effective strain
-   * @param rom_effective_strain_derivative Derivative of ROM calculated effective strain with
+   * @param drom_effective_strain_dstress Derivative of ROM calculated effective strain with
    *respect to stress
+   * @return flag that indicates ROM was skipped
    */
-
-  void computeROMStrainRate(const Real dt,
-                            const Real & mobile_dislocations_old,
-                            const Real & immobile_dislocations_old,
-                            const ADReal & trial_stress,
-                            const Real & effective_strain_old,
-                            const ADReal & temperature,
-                            const ADReal & environmental,
-                            ADReal & mobile_dislocation_increment,
-                            ADReal & immobile_dislocation_increment,
-                            ADReal & rom_effective_strain,
-                            ADReal & rom_effective_strain_derivative);
+  ADReal computeROMStrainRate(const unsigned out_idx, const bool jacobian = false);
 
   /**
    * Function to check input values against applicability windows set by ROM data set
-   * @param input Vector of input values
+   * @param input value
+   * @return flag to indicate to continue computing ROM
    */
-  void checkInputWindows(std::vector<ADReal> & input);
+  ADReal checkInputWindow(ADReal & input,
```

Comment:
...and you need to add some here. Same goes for all the other functions with changed signatures.

Reply:
... and these